site/en/docs/extensions/mv3/match_patterns/index.md

```diff
@@ -3,154 +3,66 @@ layout: 'layouts/doc-post.njk'
 title: 'Match patterns'
 seoTitle: 'Chrome Extensions match patterns'
 date: 2012-09-18
-updated: 2023-07-28
+updated: 2023-08-31
 description: Understanding URL match patterns in Chrome extensions.
 ---
 
-## Overview {: #overview }
-
-Extensions use match patterns to specify a group of URLs. They are mainly used to declare [host permissions][host-permissions] along with other permissions to use certain Chrome APIs or when injecting [content scripts][content-scripts]. Extensions also use match patterns in:
-
-- [`"web_accessible_resources.matches"`][war]
-- [`"externally_connectable.matches"`][ext-connect]
-
-## What is a match pattern? {: #what }
-
-A match pattern is a URL that begins with a permitted scheme, separated by a `://`, followed by a host and path. It can contain wildcard (`*`) characters. Most match patterns are composed of three parts:
+A match pattern is a URL with the following structure, used to specify a group of URLs:
 
 ```text
 &lt;scheme&gt;://&lt;host&gt;/&lt;path&gt;
 ```
 
-Each part can use wildcards `*`. Below is a detailed description:
-
-- **scheme**: Must include a valid scheme: `'http'`, `'https'`, `'file'`, `'ftp'`, or `'urn'`. The wildcard `*` only matches `http` or `https`.
-
+- **scheme**: Must be one of `'http'`, `'https'`, `'file'`, `'ftp'`, or `'urn'`. The wildcard `*` only matches `http` or `https`.
+  - For information on injecting content scripts into unsupported schemes, such as `about:` and `data:`, see [Injecting in related frames][cs-frames].
+  - The scheme must be separated from the rest of the pattern using a double slash (`//`).
 - **host**: A hostname (`www.example.com`), a `*` before the hostname to match subdomains (`*.example.com`), or just a wildcard `*`.
+  - If you use a wildcard in the host pattern, it must be the first or only character, and it must be followed by a period (`.`) or forward slash (`/`).
+- **path**: Must contain at least a forward slash. The slash by itself is treated as a wildcard `/*`. Other examples include the following:
+  * `/foo*`
+  * `/foo/bar`.
 
-- **path**: Must start with `/` and be present. If alone, it is treated as `/*`. For example: `/*`, `/foo*`, or `/foo/bar`. Each '`*`' matches 0 or more characters.
+Extensions use match patterns for a variety of purposes, including declaring [host permissions][host-permissions] and other permissions for Chrome APIs or [content script][content-scripts] injection, as well as [`"web_accessible_resources.matches"`][war] and [`"externally_connectable.matches"`][ext-connect].
```

Comment:
This seems like it should be a bulleted list. 

@oliverdunk @AmySteam Are we missing any places where match patterns can be used?

Reply:
I can't think of any others - thanks for checking!

Reply:
To clean up the bullet list structure: what do WAR and Externally Connectable do with the match patterns? It sounds to me like those aren't part of the declaring permissions part of the sentence.

Reply:
The match patterns are for URLs. They effectively grant access permission to URLs.

Reply:
Updated. How's this?

Reply:
That looks good. Can you make it a bulleted list?

Reply:
## Original

So originally I separated the permissions part with the "Other ways that extensions use match patterns".

<img width="733" alt="Screenshot 2023-09-08 at 16 28 28" src="https://github.com/GoogleChrome/developer.chrome.com/assets/37001393/d4557bbe-4619-4fe3-b264-5c67981e5cc4">
 
## New approach

The new approach groups it all together as "permissions", which is not accurate.

<img width="623" alt="Screenshot 2023-09-08 at 16 27 37" src="https://github.com/GoogleChrome/developer.chrome.com/assets/37001393/adad857d-a88a-4f36-81b6-ae60de63c43f">

## Explanation

The match patterns under **web accessible resources** declare which websites can access specific resources that are bundled with the Chrome extension.

The match patterns under "externally_connectable" declare which websites can send and receive messages from the Chrome extension. 

These last two use cases IMO don't have anything to do with permissions.

## What problem are we solving?

I think the original version was clear enough without the need of a rewrite. I'm still confused about what problem we are solving with rewriting this section

Reply:
That was my question.
I was asked to make that long sentence into a bulleted list to make it easier to read. To do that, I needed to understand  how all the use cases related to each other, and specifically whether those last two use cases had anything to do with permissions. I misinterpreted the previous answer to that question as "yes, these are all about permissions," which is how I ended up with that list.

I'd still like to understand what WAR and Externally Connectable are and how they use match patterns, but I don't need that information to fix the list.

Reply:
**Match pattern use explanation**

- The match patterns under web accessible resources declare which URLs are allowed to access specific resources that are bundled with the Chrome extension.

- The match patterns under "externally_connectable" declare which URLS are allowed to send and receive messages from the Chrome extension.

Reply:
So is it useful to say, for example, "Granting access to [web-accessible resources]" and "Allowing message sending and receiving using [`"externally_connectable.matches"`]" in the list?

Reply:
> So is it useful to say, for example, "Granting access to [web-accessible resources]" and "Allowing message sending and receiving using [`"externally_connectable.matches"`]" in the list?

This sounds like a good idea 👍.

Reply:
> > So is it useful to say, for example, "Granting access to [web-accessible resources]" and "Allowing message sending and receiving using [`"externally_connectable.matches"`]" in the list?
> 
> This sounds like a good idea 👍.

Done in the new commit.